### PR TITLE
Implemented coverband to expose unused code

### PR DESCRIPTION
**Pull request by RachalCassity**
### Do not merge until the [devops PR](https://github.com/department-of-veterans-affairs/devops/pull/11656) and the [ Infra PR](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/665) have been deployed.

There is code in production that is not in use. We should know what is and isn't being called. [Coverband](https://github.com/danmayer/coverband) provides metrics on code that is called in production.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#29895

### CODE REVIEW

File: config/coverband.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# config/coverband.rb NOT in the initializers
+Coverband.configure do |config|
+  config.store = Coverband::Adapters::RedisStore.new(Redis.new(url: Settings.redis.app_data.secondary_url))
```
Comment (ericboehs) [inline config/coverband.rb:5]:
```suggestion
  config.store = Coverband::Adapters::RedisStore.new($redis)
```

I don't know if the secondary url will even be set here. App data is the right spot but $redis will already be initialized with a Redis instance.

File: config/routes.rb [hunk 2]
```diff
@@ -415,15 +415,18 @@
   require 'sidekiq/pro/web' if Gem.loaded_specs.key?('sidekiq-pro')
   require 'sidekiq-ent/web' if Gem.loaded_specs.key?('sidekiq-ent')
   require 'github_authentication/sidekiq_web'
+  require 'github_authentication/coverband_reporters_web'
 
   mount Sidekiq::Web, at: '/sidekiq'
 
-  Sidekiq::Web.register GithubAuthentication::SidekiqWeb unless Rails.env.development? || Settings.sidekiq_admin_panel
+  Sidekiq::Web.register GithubAuthentication::SidekiqWeb
```
Comment (ericboehs) [inline config/routes.rb:422]: Was this moved somewhere?

File: config/settings.yml [hunk 3]
```diff
@@ -1370,3 +1370,10 @@ form1095_b:
     bucket: 'bucket'
     enabled: true
     region: 'region'
+
+coverband:
```
Comment (ericboehs) [inline config/settings.yml:1374]: Is there a corresponding devops PR for this?

File: config/routes.rb (hunk 2 above)
Comment (RachalCassity) [reply config/routes.rb:422]: Whoops, I added this for testing.

File: config/coverband.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# config/coverband.rb NOT in the initializers
+Coverband.configure do |config|
+  config.store = Coverband::Adapters::RedisStore.new(Redis.new($redis)
```
Comment (ericboehs) [inline config/coverband.rb:5]:
```suggestion
  config.store = Coverband::Adapters::RedisStore.new(Redis.new($redis))
```

File: config/initializers/warden_github.rb [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 module WardenGithubStrategyExtensions
   def authenticate!
-    if scope == :sidekiq && session[:sidekiq_user].present?
-      success!(session[:sidekiq_user])
+    if (scope == :sidekiq && session[:sidekiq_user].present?) ||
+       (scope == :coverband && session[:coverband_user].present?)
```
Comment (LindseySaari) [inline config/initializers/warden_github.rb:6]: Do you need to put this success!(session[:sidekiq_user]) line back?

Comment (RachalCassity) [reply config/initializers/warden_github.rb]: @LindseySaari Thank you. I had to remove a lot for dev testing. Thank you!

File: spec/lib/github_authentication/coverband_reports_web_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
```
Comment (LindseySaari) [inline spec/lib/github_authentication/coverband_reports_web_spec.rb:3]: I think this needs to be renamed to `coverband_reporters_web` right now it's reports

**Review by LindseySaari — APPROVED** [review]
Awesome 👍

**Review by LindseySaari — COMMENTED** [review]
It looks like you'll have to put in a corresponding PR to update the K8s secrets to include the new parameter store values: https://github.com/department-of-veterans-affairs/devops/pull/11656#issuecomment-1194306297

**Pull request merged**